src/System.IO.Pipelines.Networking.Sockets/SocketConnection.cs

```diff
@@ -351,7 +351,7 @@ private enum BufferStyle
                         // before flushing (entirely arbitrarily) - might want to make this configurable later
 
                         haveWriteBuffer = true;
-                        int FlushInputEveryBytes = 4 * Pool.MaxBufferSize;
+                        int FlushInputEveryBytes = 24;
```

Comment:
Can you please explain why this change was needed?

Reply:
Yeah, it's a random flush size computation in the sample and it overflowed when `Pool.MaxBufferSize` is maxint